Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/UI/LeaderBoard.hs

```diff
@@ -10,15 +10,23 @@ import Servant
 import Tools.Auth
 
 type API =
-  "driver"
-    :> "leaderBoard"
-    :> TokenAuth
-    :> QueryParam "limit" Integer
-    :> Get '[JSON] DLeaderBoard.LeaderBoardRes
+  "driver" :> "leaderBoard"
+    :> ( TokenAuth
+           :> QueryParam "limit" Integer
+           :> Get '[JSON] DLeaderBoard.LeaderBoardRes
+           :<|> TokenAuth
+           :> QueryParam "limit" Int
+           :> ReqBody '[JSON] DLeaderBoard.LeaderBoardReq
+           :> Post '[JSON] DLeaderBoard.LeaderBoardRes
+       )
 
 handler :: FlowServer API
 handler =
   getDriverLeaderBoard
+    :<|> getLeaderBoard
```

Comment:
'Daily' keyword can be added in fn name, the diff b/w these 2 isn't clear without that